crypto/params.c

```diff
@@ -16,8 +16,10 @@ OSSL_PARAM *OSSL_PARAM_locate(OSSL_PARAM *p, const char *key)
 {
     if (p != NULL && key != NULL)
         for (; p->key != NULL; p++)
-            if (strcmp(key, p->key) == 0)
+            if (strcmp(key, p->key) == 0) {
+                OSSL_PARAM_set_used(p, 1);
```

Comment:
I see no reason for this call, since you are now setting the flag in `OSSL_PARAM_get_int` etc

Reply:
Is it implausible for a receiver to use locate to find the param and then twiddle the fields directly?

I'm fine either way BTW.

Reply:
Thinking about this overnight, I concur.  The locate shouldn't set the flag.

I'm almost tempted to define the flags:

1. located
2. read
3. written

A get function sets the read flag, a set function sets the written and locate sets the located.